pkg/github/github_manager.go

```diff
@@ -0,0 +1,99 @@
+package github
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/ghodss/yaml"
+	klum "github.com/jadolg/klum/pkg/apis/klum.cattle.io/v1alpha1"
+	log "github.com/sirupsen/logrus"
+)
+
+func UploadGithubSecret(userSync *klum.UserSync, kubeconfig *klum.Kubeconfig, githubURL string, githubToken string) error {
+	githubSync := userSync.Spec.Github
+	if !githubSync.Valid() {
+		return fmt.Errorf("not enough github data to be able to create a GitHub secret")
+	}
+
+	kubeconfigYAML, err := toYAMLString(kubeconfig.Spec)
+	if err != nil {
+		return err
+	}
+
+	kubeconfigYAMLb64 := base64.StdEncoding.EncodeToString([]byte(kubeconfigYAML))
+
+	latestKubeconfigUploaded, present := userSync.Annotations["klum.cattle.io/lastest.upload.github"]
+	if present && latestKubeconfigUploaded == kubeconfigYAMLb64 {
+		return nil
+	}
+	userSync.Annotations["klum.cattle.io/lastest.upload.github"] = kubeconfigYAMLb64
```

Comment:
I think that shouldn't be overwritten :thinking: 
In any case, what I want to do is stop the reconciliation of the object on time. It shouldn't really matter where I put the data as long as I can compare it in the next loop cycle.
I changed the data so it's a hash instead of base64 now and would like to keep it as an annotation.
I wanted to mimic the behaviour of `kubectl.kubernetes.io/last-applied-configuration`

Reply:
But doing hash is fine

Reply:
Users can be enabled/disabled and this generates a new secret and new kubeconfig.